### PR TITLE
Highlight for annotated variables

### PR DESCRIPTION
**Pull request by evaliu2002**
VSCode Issue 60 Report

Features Finished
- Visual Indication for Annotated Variables
  - Used Decorators to highlight the variables in blue
- Context (Right Click Popup) Menu Button to Turn the Visual Indications On and Off
- Extension User Settings Checkbox Option to Turn the Visual Indications On and Off
- Syncing of Context Menu and Extension User Settings
- Visual Indication Color Changer

Concerns/Compromises
- There are two buttons, one for turning the feature on and one for turning the feature off
  - Solution: Realized that I should have used “config.” to grab the setting instead of “extension.” or the “vscode.workspace.getConfi”
- To turn the feature on using the right-click popup menu button and change the visual indication color, you must reload the entire window
  - This may not be a huge issue. Prefer to be streamlined but otherwise alright. Will talk to Dr. Hellerstein and Dr. Sauro.
- The extension user setting for the switch doesn’t sync with the right-click popup menu, it acts more like a master switch but works very well otherwise
  - Solution: Realized that I should have set the third parameter to true for global application of the update. Issue is now solved

Unit Testing
- Switch indication on for user settings
- Switch indication off for user settings
- Dynamic button for right-click menu
- Button/Setting syncing
- Visual Indication Highlight Color Change

Note:
As of VSCode 1.37.1 (2019-08-30), there is no way to write an automated test for decorations using the [decorations extensions API](https://code.visualstudio.com/api/references/vscode-api#TextEditorDecorationType).

There are automated tests of decorations in the vscode sources, particularly [modelDecorations.test.ts](https://github.com/microsoft/vscode/blob/master/src/vs/editor/test/common/model/modelDecorations.test.ts), but they use private APIs.

Microsoft has an example extension that uses decorators, [decorator-sample](https://github.com/microsoft/vscode-extension-samples/tree/master/decorator-sample), but it has no tests.

### CODE REVIEW

**Review by mastevb — DISMISSED** [review]

File: vscode-antimony/.vscode/settings.json
Comment (evaliu2002) [reply vscode-antimony/.vscode/settings.json:12]: I had a "The terminal process "/bin/zsh '-c', 'npm run watch'" failed to launch (exit code: 127)." error and the recommended solution was to set this to true. This was from https://github.com/microsoft/vscode/issues/104736. I'm not sure if this ultimately resolved the error as I tried a multitude of recommended solutions before the error stopped. Next time I will make note to document errors I encounter, the process of solving the errors and the solution.

File: vscode-antimony/tsconfig.json
Comment (evaliu2002) [reply vscode-antimony/tsconfig.json:11]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/package.json
Comment (evaliu2002) [reply vscode-antimony/package.json]: Changed and resolved

File: vscode-antimony/src/extension.ts
Comment (evaliu2002) [reply vscode-antimony/src/extension.ts]: Changed and resolved

File: vscode-antimony/src/extension.ts
Comment (evaliu2002) [reply vscode-antimony/src/extension.ts:151]: Indentation is fixed now

File: vscode-antimony/package.json
Comment (mastevb) [reply vscode-antimony/package.json]: I don't see this getting resolved.

Comment (mastevb) [reply vscode-antimony/package.json]: I don't see this getting resolved.

File: vscode-antimony/package.json
Comment (mastevb) [reply vscode-antimony/package.json]: I also don't see this getting resolved.

**Review by evaliu2002 — COMMENTED** [review]
Is this how I would submit my own replies?

File: vscode-antimony/src/server/stibium/stibium/analysis.py
Comment (evaliu2002) [reply vscode-antimony/src/server/stibium/stibium/analysis.py]: Aldrich managed to get this implementation way to work.

**Comment by evaliu2002** [comment]
Aldrich implemented the requested back end change. Awaiting second review still

File: vscode-antimony/tsconfig.json
Comment (luciansmith) [reply vscode-antimony/tsconfig.json:11]: This is not actually fixed, looks like.

Comment (luciansmith) [reply vscode-antimony/tsconfig.json:11]: Aha:  the problem is that the lines above are using tabs, while the new line is using spaces.

**Review by luciansmith — APPROVED** [review]
Great!

File: vscode-antimony/src/server/stibium/stibium/analysis.py
Comment (luciansmith) [reply vscode-antimony/src/server/stibium/stibium/analysis.py]: What's the conclusion here--does self.pending_annotations work now?

Comment (evaliu2002) [reply vscode-antimony/src/server/stibium/stibium/analysis.py]: Yes, it works now

**Pull request merged**
